packages/teleport/src/components/LabelSelector/LabelSelector.tsx

```diff
@@ -0,0 +1,273 @@
+              />
+            </div>
+            <Popover
+              id="simple-popper"
+              open={showTooltip}
+              anchorOrigin={{
+                vertical: 'bottom',
+                horizontal: 'right',
+              }}
+              transformOrigin={{
+                vertical: 'top',
+                horizontal: 'left',
+              }}
+              anchorEl={infoIconRef.current}
+              onClose={() => setShowTooltip(false)}
+            >
+              <Box
+                bg="#011223"
+                color="white"
+                width={362}
+                p={4}
+                style={{
+                  boxShadow: '0px 8px 14px rgba(12, 12, 14, 0.07)',
+                  borderRadius: '8px',
+                }}
+              >
+                Teleport provides users the ability to add labels (in the form
+                of key:value pairs) to resources. Some valid example labels are
+                “env: prod” and “arch: x86_64”. Labels, used in conjunction with
+                roles, define access in Teleport. For example, you can specify
+                that users with the “on-call” role can access resources labeled
+                “env: prod”. For more information, check out our documentation
+                on{' '}
+                <Link
+                  href="https://goteleport.com/docs/setup/admin/trustedclusters/"
+                  target="_blank"
+                >
+                  RBAC
+                </Link>{' '}
+                and{' '}
+                <Link
+                  href="https://goteleport.com/docs/setup/admin/labels/"
+                  target="_blank"
+                  rel="noreferrer"
+                >
+                  labels
+                </Link>
+                .
+              </Box>
+            </Popover>
+          </Flex>
+          <Text>
+            <Link
+              href="https://goteleport.com/docs/setup/admin/labels/"
+              target="_blank"
+              color="rgb(255, 255, 255)"
+            >
+              View Documentation
+            </Link>
+          </Text>
+        </Flex>
+      </Heading>
+      <LabelContainer
+        onClick={() => setShowAdd(!showAdd)}
+        data-testid="label-container"
+      >
+        {labels.length === 0 && (
+          <Text color="rgba(255, 255, 255, 0.1)">Click to add new labels.</Text>
+        )}
+        {labelList({ labels, onDismiss: handleLabelDismiss })}
+      </LabelContainer>
+      {showAdd && (
+        <AddLabelContainer
+          ref={addLabelContainerRef}
+          data-testid="add-label-container"
+        >
+          <AddLabelInput
+            name="addLabel"
+            value={newLabel}
+            onChange={e => {
+              setNewLabel(e.target.value);
+            }}
+            onKeyPress={e => {
+              // Add a new label on `Enter` if it's valid.
+              if (e.charCode === 13 && validLabel) {
```

Comment:
looks like KeyboardEvent.charCode is also deprecated now? https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/charCode

Reply:
good catch, updated